### ISSUE TITLE
Core tap stops updating

### ISSUE THREAD

**Issue by chrisfinazzo** [Homebrew/brew#2357]
# Please follow the general troubleshooting steps first:

- [x] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed as many issues as possible and retried your prior step?
- [x] Confirmed this is a problem with Homebrew/brew and not specific formulae? If it's a formulae-specific problem please file this issue at https://github.com/Homebrew/homebrew-core/issues/new

### Bug reports:

Updates to outdated formulae are not being applied. `brew outdated` will initially report that there is a new version of a formula, but `brew upgrade` returns to the prompt without actually doing anything. Subsequent use of `brew outdated` produces no output.

    $ brew config
    HOMEBREW_VERSION: 1.1.11-100-ga9cd74de5
    ORIGIN: https://github.com/Homebrew/brew.git
    HEAD: a9cd74de59f4825555efe0c161a62d6306a1b768
    Last commit: 16 minutes ago
    Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
    Core tap HEAD: 6f3003529004041e0d78bfb7fa0ebcd60a4ab3a5
    Core tap last commit: 6 weeks ago
    HOMEBREW_PREFIX: /usr/local
    HOMEBREW_REPOSITORY: /usr/local/Homebrew
    HOMEBREW_CELLAR: /usr/local/Cellar
    HOMEBREW_BOTTLE_DOMAIN: https://homebrew.bintray.com
    CPU: octa-core 64-bit haswell
    Homebrew Ruby: 2.0.0-p648
    Clang: 7.3 build 703
    Git: 2.11.1 => /usr/local/bin/git
    Perl: /usr/bin/perl
    Python: /usr/bin/python
    Ruby: /Users/Chris/.rbenv/shims/ruby => /usr/local/Cellar/ruby/2.4.0/bin/ruby
    Java: N/A
    macOS: 10.11.6-x86_64
    Xcode: 7.3.1
    CLT: 7.3.1.0.1.1461711523
    X11: N/A

<img width="585" alt="screen shot 2017-03-17 at 10 21 17 am" src="https://cloud.githubusercontent.com/assets/1534882/24047266/8e5cf6e8-0afb-11e7-854a-4efb5bb0fbd5.png">

Seems like a reoccurance of #688/#706.

**Comment by ilovezfs** [comment]
Did you run brew update?

**Comment by chrisfinazzo** [comment]
@ilovezfs 

Yes, I did.

    $ brew update
    Updated 1 tap (caskroom/cask).
    No changes to formulae.

**Comment by ilovezfs** [comment]
OK. And now what do `brew config` and `brew outdated say?

**Comment by chrisfinazzo** [comment]
$ brew config
    HOMEBREW_VERSION: 1.1.11-100-ga9cd74de5
    ORIGIN: https://github.com/Homebrew/brew.git
    HEAD: a9cd74de59f4825555efe0c161a62d6306a1b768
    Last commit: 56 minutes ago
    Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
    Core tap HEAD: 6f3003529004041e0d78bfb7fa0ebcd60a4ab3a5
    Core tap last commit: 6 weeks ago
    HOMEBREW_PREFIX: /usr/local
    HOMEBREW_REPOSITORY: /usr/local/Homebrew
    HOMEBREW_CELLAR: /usr/local/Cellar
    HOMEBREW_BOTTLE_DOMAIN: https://homebrew.bintray.com
    CPU: octa-core 64-bit haswell
    Homebrew Ruby: 2.0.0-p648
    Clang: 7.3 build 703
    Git: 2.11.1 => /usr/local/bin/git
    Perl: /usr/bin/perl
    Python: /usr/bin/python
    Ruby: /Users/Chris/.rbenv/shims/ruby => /usr/local/Cellar/ruby/2.4.0/bin/ruby
    Java: N/A
    macOS: 10.11.6-x86_64
    Xcode: 7.3.1
    CLT: 7.3.1.0.1.1461711523
    X11: N/A

**Comment by ilovezfs** [comment]
```
git -C `brew --repo homebrew/core` status
```

**Comment by chrisfinazzo** [comment]
![UGH](https://media.giphy.com/media/14aUO0Mf7dWDXW/giphy.gif).

Tl;dr Branches are a thing and when you have uncommitted stuff there `master` doesn't know what's up.

I assume the behavior of `git stash` is the same even when you are in a Core repo directory?

**Comment by ilovezfs** [comment]
Yeah stash is relative to whichever git repository your current working directory is in (unless you also passed -C)
